### PR TITLE
[Clang] Fix passing `-offload-compress` when compiling and linking device images separately

### PR DESCRIPTION
**Pull request by uditagarwal97**
When compiling and linking device images separately, compression related flags are not propagated correctly to `offload-wrapper`. This PR fixes that and updates the test accordingly.

### CODE REVIEW

**Comment by uditagarwal97** [comment]
> The wrapping that occurs here is unique for FPGA when working with intermediate AOCR and AOCX files. I'm not sure how this would interact with consumption of the host object.
> 
> There are tests in test-e2e/AOT/fpga-aoc-archive* that exercise this behavior where `-offload-compress` can be added to see the impact. Can those be verified? If those don't take kindly to the additional compression steps, the compression options will have to be added to non `-fintelfpga` only.

True - I've updated the PR to only pass compression options to non-FPGA targets.

**Review by aelovikov-intel — APPROVED** [review]
Everything here is driver-specific, Mike's approval should be enough and mine is a pure formality.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -10350,33 +10350,50 @@ void OffloadWrapper::ConstructJob(Compilation &C, const JobAction &JA,
...
 
-    // Add offload targets and inputs.
-    CmdArgs.push_back(C.getArgs().MakeArgString(
-        Twine("-kind=") + Action::GetOffloadKindName(OffloadingKind)));
-    CmdArgs.push_back(
-        TCArgs.MakeArgString(Twine("-target=") + Triple.getTriple()));
+    // When compiling and linking separately, we need to propagate the
+    // compression related CLI options to offload-wrapper. Don't propagate
+    // these options when wrapping objects for FPGA.
+    if (!C.getArgs().hasArg(options::OPT_fintelfpga)) {
```
Comment (bader) [inline clang/lib/Driver/ToolChains/Clang.cpp:10359]: Code style nit.
When if-statement have both statement-true and statement-false, the code more readable when if-condition doesn't use negation.
I.e. prefer
```c++
if (cond) 
  A;
else
  B;
```
to
```c++
if (!cond) 
  B;
else
  A;
```

Comment (uditagarwal97) [reply clang/lib/Driver/ToolChains/Clang.cpp]: Fixed in 20b31b75a96a37ee56ae279f79bfdfd4bb1a0388

File: sycl/test-e2e/Compression/compression_separate_compile.cpp [hunk 2]
```diff
@@ -10,6 +10,13 @@
 //////////////////////   Link device images
 // RUN: %clangxx --offload-compress -fsycl -fsycl-link -fsycl-targets=spir64_x86_64 -fPIC %t_kernel1_aot.o %t_kernel2_aot.o -o %t_compressed_image.o -v
 
+// Make sure the clang-offload-wrapper is called with the --offload-compress
+// option.
+// RUN: %clangxx --offload-compress -fsycl -fsycl-link -fsycl-targets=spir64_x86_64 -fPIC %t_kernel1_aot.o %t_kernel2_aot.o -o %t_compressed_image.o -### &> %t_driver_opts.txt
+// RUN: FileCheck -input-file=%t_driver_opts.txt %s --check-prefix=CHECK-DRIVER-OPTS
+
+// CHECK-DRIVER-OPTS: clang-offload-wrapper{{.*}} "-offload-compress"
```
Comment (AlexeySachkov) [inline sycl/test-e2e/Compression/compression_separate_compile.cpp:18]: Why is this in E2E tests, when its a pure clang driver test and pure clang driver change? The test seems to be misplaced to me

**Pull request merged**